Use buildbot GitHub status integration

aneeshusa opened the pull request:
Buildbot 0.8.12 has a GitHub status integration; it would be better to
use that directly instead of making Homu act as a middleman. Note that
this commit does not change the Homu configuration, but only adds a
GitHub status integration for Buildbot.

Note that this requires a new secret value in the Pillar - a token
that Buildbot can use to access the GitHub status API.

See http://docs.buildbot.net/current/manual/cfg-statustargets.html#githubstatus
for more details on the integration.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/saltfs/387)
<!-- Reviewable:end -->

bors-servo commented:
:umbrella: The latest upstream changes (presumably #410) made this pull request unmergeable. Please resolve the merge conflicts.

edunham commented:
This looks like it might report more builds than Homu is doing, which would be an improvement. 

@bors-servo r+

Sorry for the delay in review, but this looks good to go after fixing the merge conflict.

bors-servo commented:
:pushpin: Commit 3964299 has been approved by `edunham`

<!-- @bors-servo r=edunham 39642992a267423f5c053f55fe02823414e679e8 -->

bors-servo commented:
:lock: Merge conflict

aneeshusa commented:
Rebased. @bors-servo r=edunham

Fun fact: I completely forgot I had already removed Buildbot categories as the first commit in this PR and opened #412 recently as an E-easy, hence only one commit left.

bors-servo commented:
:pushpin: Commit 554165b has been approved by `edunham`

<!-- @bors-servo r=edunham 554165b36e9b3147dd89ee34c6188eacee343aa8 -->

bors-servo commented:
:hourglass: Testing commit 554165b with merge 8b2cc91...

bors-servo commented:
:bulb: This pull request was already approved, no need to approve it again.

- This pull request is currently being tested. If there's no response from the continuous integration service, you may use `retry` to trigger a build again.

bors-servo commented:
:pushpin: Commit 554165b has been approved by `edunham`

<!-- @bors-servo r=edunham 554165b36e9b3147dd89ee34c6188eacee343aa8 -->

bors-servo commented:
:sunny: Test successful - [travis](https://travis-ci.org/servo/saltfs/builds/144068867)
